src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,105 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+
+        long distanceToX = Math.abs(robot.getX() - (long) toX);
+        long distanceToY = Math.abs(robot.getY() - (long) toY);
+
+        if (robot.getX() < toX) {
```

Comment:
Separate this blocks to smaller methods and have a think, how to use them properly. It's a good practice to have some small methods instead of single and giant :)

Reply:
To solve this I've asked my friend  for some advice ) So it's not 100% my solution